include/zephyr/drivers/pwm/max31790.h

```diff
@@ -0,0 +1,32 @@
+/*
+ * Copyright (c) 2023 SILA Embedded Solutions GmbH
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#ifndef ZEPHYR_INCLUDE_DRIVERS_PWM_MAX31790_H_
+#define ZEPHYR_INCLUDE_DRIVERS_PWM_MAX31790_H_
+
+#define PWM_MAX31790_FLAG_RPM_MODE_POS		 8
+#define PWM_MAX31790_FLAG_SPEED_RANGE_POS	 9
+#define PWM_MAX31790_FLAG_PWM_RATE_OF_CHANGE_POS 12
+#define PWM_MAX31790_FLAG_SPIN_UP_POS		 15
+
+#define PWM_MAX31790_FLAG_RPM_MODE	       (1 << PWM_MAX31790_FLAG_RPM_MODE_POS)
+#define PWM_MAX31790_FLAG_SPEED_RANGE_1	       (0b000 << PWM_MAX31790_FLAG_SPEED_RANGE_POS)
+#define PWM_MAX31790_FLAG_SPEED_RANGE_2	       (0b001 << PWM_MAX31790_FLAG_SPEED_RANGE_POS)
+#define PWM_MAX31790_FLAG_SPEED_RANGE_4	       (0b010 << PWM_MAX31790_FLAG_SPEED_RANGE_POS)
+#define PWM_MAX31790_FLAG_SPEED_RANGE_8	       (0b011 << PWM_MAX31790_FLAG_SPEED_RANGE_POS)
+#define PWM_MAX31790_FLAG_SPEED_RANGE_16       (0b100 << PWM_MAX31790_FLAG_SPEED_RANGE_POS)
+#define PWM_MAX31790_FLAG_SPEED_RANGE_32       (0b101 << PWM_MAX31790_FLAG_SPEED_RANGE_POS)
+#define PWM_MAX31790_FLAG_PWM_RATE_OF_CHANGE_0 (0b000 << PWM_MAX31790_FLAG_PWM_RATE_OF_CHANGE_POS)
+#define PWM_MAX31790_FLAG_PWM_RATE_OF_CHANGE_1 (0b001 << PWM_MAX31790_FLAG_PWM_RATE_OF_CHANGE_POS)
+#define PWM_MAX31790_FLAG_PWM_RATE_OF_CHANGE_2 (0b010 << PWM_MAX31790_FLAG_PWM_RATE_OF_CHANGE_POS)
+#define PWM_MAX31790_FLAG_PWM_RATE_OF_CHANGE_3 (0b011 << PWM_MAX31790_FLAG_PWM_RATE_OF_CHANGE_POS)
+#define PWM_MAX31790_FLAG_PWM_RATE_OF_CHANGE_4 (0b100 << PWM_MAX31790_FLAG_PWM_RATE_OF_CHANGE_POS)
+#define PWM_MAX31790_FLAG_PWM_RATE_OF_CHANGE_5 (0b101 << PWM_MAX31790_FLAG_PWM_RATE_OF_CHANGE_POS)
+#define PWM_MAX31790_FLAG_PWM_RATE_OF_CHANGE_6 (0b110 << PWM_MAX31790_FLAG_PWM_RATE_OF_CHANGE_POS)
+#define PWM_MAX31790_FLAG_PWM_RATE_OF_CHANGE_7 (0b111 << PWM_MAX31790_FLAG_PWM_RATE_OF_CHANGE_POS)
+#define PWM_MAX31790_FLAG_SPIN_UP	       (1 << PWM_MAX31790_FLAG_SPIN_UP_POS)
+
+#endif /* ZEPHYR_INCLUDE_DRIVERS_PWM_MAX31790_H_ */
```

Comment:
Because these flags can be set for pwm_set_cycles in pwm_flags_t. As the documentation of pwm_flags_t states that the upper 8 bits are reserved for SoC specific usage, I used them to implement the RPM mode and also provide additional configuration to the user.